api/v1alpha1/cdapmaster_types.go

```diff
@@ -98,6 +107,14 @@ type CDAPServiceSpec struct {
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
 	// Env is a list of environment variables for the master service container.
 	Env []corev1.EnvVar `json:"env,omitempty"`
+	// ConfigMapVolumes defines a map from ConfigMap names to volume mount path for this service
+	// Key is the configmap object name. Value is the mount path.
+	// This adds ConfigMap data to the directory specified by the volume mount path.
+	ConfigMapVolumes map[string]string `json:"configMapVolumes,omitempty"`
+	// SecretVolumes defines a map from Secret names to volume mount path for this service
+	// Key is the secret object name. Value is the mount path.
+	// This adds Secret data to the directory specified by the volume mount path.
+	SecretVolumes map[string]string `json:"secretVolumes,omitempty"`
```

Comment:
Are these used for overriding those settings in CDAPMasterSpec ? If so, mind updating the comment to state that explicitly?

If I recall correctly, ConfigMapVolumes defined in CDAPMasterSpec will mount config map to all statefulsets and deployments.

Reply:
It is not for overriding, but appending/merging.